### PR TITLE
New ecms tutorials

### PR DESCRIPTION
**Pull request by AnnaWiniwarter**
This branch contains the EC-MS tutorial currently published on the ixdat documentation page (i.e. relying only on ixdat functions). Can be merged ASAP.

### CODE REVIEW

File: ec_ms_quantification/EC-MS_ixdat_tutorial.ipynb
Comment (ScottSoren) [inline ec_ms_quantification/EC-MS_ixdat_tutorial.ipynb:109]: Could you import all of them at once here with `Measurement.read_set()` or at least mention it as a possibility?

File: ec_ms_quantification/EC-MS_ixdat_tutorial.ipynb
Comment (ScottSoren) [inline ec_ms_quantification/EC-MS_ixdat_tutorial.ipynb:13]: Include a link to the data on Zenodo. It's in the tutorials repo's README but would be nice to have here as well :)

File: ec_ms_quantification/EC-MS_ixdat_tutorial.ipynb
Comment (ScottSoren) [inline ec_ms_quantification/EC-MS_ixdat_tutorial.ipynb:149]: Nice! But a question of how focused the tutorial should be. I also show importing / exporting in the [spectroelectrochemistry tutorial](https://github.com/ixdat/tutorials/blob/main/L2_techniques/spectroelectrochemistry/spectroelectrochemistry_demo.ipynb). It's also in [the L3 tutorial](https://github.com/ixdat/tutorials/blob/main/L3_data_structure/01_reading_and_using_data.ipynb). Maybe it actually belongs in the top, L1 "getting started" layer? 

An advantage of very focused technique tutorials is that they could start with ixdat-exported files small enough to host directly here on github.

Fine now to have it multiple places and fine-tune later if someone complains the tutorials aren't focused enough...

Also if the point is file size reduction, consider using the `time_step` argument.

File: ec_ms_quantification/EC-MS_ixdat_tutorial.ipynb
Comment (ScottSoren) [inline ec_ms_quantification/EC-MS_ixdat_tutorial.ipynb]: Consider this instead:
`meas.tstamp += meas["M2"].t[0] + 1700`
Then it is fine to run multiple times.

File: ec_ms_quantification/EC-MS_ixdat_tutorial.ipynb
Comment (ScottSoren) [inline ec_ms_quantification/EC-MS_ixdat_tutorial.ipynb]: Here I see https://github.com/ixdat/ixdat/issues/118 in play.
Consider for now using:
`axes_a = meas.plot_measurement(tspan=(0, 3200), J_name="raw_current")`
Then it works out fine :)

File: ec_ms_quantification/EC-MS_ixdat_tutorial.ipynb
Comment (ScottSoren) [inline ec_ms_quantification/EC-MS_ixdat_tutorial.ipynb]: Does it need to be implemented? Easy if so. But rate is just integral divided by time interval, so as long as the time intervals are the same there is no difference, I think?

File: ec_ms_quantification/EC-MS_ixdat_tutorial.ipynb
Comment (ScottSoren) [inline ec_ms_quantification/EC-MS_ixdat_tutorial.ipynb]: I see you've switched to the pre-save-and-reload `meas_combi`. That is confusing. But I can also see why you had to do so - `selector` does not get saved in the zilien EC-MS data. 

A way to avoid the confusion would be the following refactor:
`meas` --> `meas_zilien`
`meas_combi` --> `meas`
And then just use `meas` after the part about importing.

An even better solution in my opinion, if you have time: spin off the top part of this tutorial into a tutorial on the Zilien reader (which covers combining with biologic and exporting the data) and then start this tutorial from ixdat-formatted .csv files, so that it is equipment-agnostic. We could also get to this later if we decide to make a complete readers tutorials "layer".

File: ec_ms_quantification/EC-MS_ixdat_tutorial.ipynb
Comment (ScottSoren) [inline ec_ms_quantification/EC-MS_ixdat_tutorial.ipynb]: Note that if you follow the refactor suggested above, you'll need to either adjust the tspans or shift it back here:
`meas.tstamp += meas_combi["M2"].t[0]`  
And, again, consider, to get the plot of the integrals right:
`axes_a2 = meas.plot_measurement(J_name="raw_current")`

File: ec_ms_quantification/EC-MS_ixdat_tutorial.ipynb
Comment (ScottSoren) [inline ec_ms_quantification/EC-MS_ixdat_tutorial.ipynb:505]: If we follow through on deprecating ixdat's `MSInlet`, this part will need to be done with `spectro_inlets_quantification` (siq).
That might require splitting the tutorial up further into an ixdat native calibration that only includes EC-MS calibration, and a siq one.

Another possibility is to define a template for `MSInlet`s in `ixdat` which interfacing packages like siq should follow (they would just need to have a `get_n_dot_0` method) so that they can be used in ixdat native calibrations, by passing an instance into `MSMeasurement.gas_flux_calibration_curve` and the like.

File: ec_ms_quantification/EC-MS_ixdat_tutorial.ipynb
Comment (ScottSoren) [inline ec_ms_quantification/EC-MS_ixdat_tutorial.ipynb:944]: Nice! :) 
But you don't actually plot calibrated data here, and it might not be relevant for MS quantification. So perhaps this would also belong better elsewhere like a spun-off `zilien` reader tutorial.

File: ec_ms_quantification/EC-MS_ixdat_tutorial.ipynb
Comment (ScottSoren) [inline ec_ms_quantification/EC-MS_ixdat_tutorial.ipynb]: I think the main point here is that it *is* possible to mix calibrated and non-calibrated data, although the user might need an explanation of why they are being shown this, like:

"Note that it is possible to mix calibrated and uncalibrated data by calling the plotting function twice. This enables one to plot the uncalibrated carrier gas signal (which may carry information about how the experiment was done even if it is not itself quantified) on one y-axis and the product signal on the other y-axis."

And it *is* possible to have one of the y-axes be logplot and one linear. Also, a better unit could be used and I think left and right are switched in the comments. Here all that is corrected:
```
axes_c = meas_c.plot_measurement(
    mol_lists = [[], ["H2", "O2"]], # right y-axis
    unit="pmol/s",
    logplot=False,
    legend=False,
)
meas_c.plot_measurement(
    mass_list=["M4", "M28", "M40"],  # left y-axis
    logplot=True,
    legend=False,
    axes=axes_c, #to reuse the axis for co-plotting two separate parts of the measurement
    #linestyle="--",
)
axes_c[2].set_ylabel("cal. sig. / [nmol s$^{-1}$]")
# axes_c[-1].set_ylim(bottom=-5)
```

File: ec_ms_quantification/EC-MS_ixdat_tutorial.ipynb
Comment (ScottSoren) [inline ec_ms_quantification/EC-MS_ixdat_tutorial.ipynb:541]: To be uncommented after solving
https://github.com/ixdat/ixdat/issues/94

File: ec_ms_quantification/EC-MS_ixdat_tutorial.ipynb
Comment (ScottSoren) [inline ec_ms_quantification/EC-MS_ixdat_tutorial.ipynb]: This could be its own markdown and code blocks. You could mention that the calibration is human-readable in notepad.

**Review by ScottSoren — CHANGES_REQUESTED** [review]
Hi Anna, 

Looks great!

I've made comments here.

The biggest one is that it might be more than one tutorial sneaked into one. I think a way to get the layer-structure separation right would be to split into a reader tutorial about zilien data and its quirks such as combining with biologic data, and which exports ixdat .csv's and then an MS calibration one which starts with those exported files. But that could be later.
Otherwise some small code suggestions and requests in a few places for more explanatory text.
I'll copy it over as is so that I can open a PR fixing the tutorials on readthedocs, but will update that after this is merged.

File: ec_ms_quantification/EC-MS_ixdat_tutorial.ipynb
Comment (AnnaWiniwarter) [reply ec_ms_quantification/EC-MS_ixdat_tutorial.ipynb:149]: I agree, eventually it should be split up into (at least) three parts, as hinted at in the intro to the tutorial. However, with the other changes to the tutorials coming up, I suggest we do this later on.

Comment (AnnaWiniwarter) [reply ec_ms_quantification/EC-MS_ixdat_tutorial.ipynb:149]: I will not prioritize file size reduction for now. Since the dataset is available from Zenodo, I think this is really a minor concern.

File: ec_ms_quantification/EC-MS_ixdat_tutorial.ipynb
Comment (AnnaWiniwarter) [reply ec_ms_quantification/EC-MS_ixdat_tutorial.ipynb]: I see only one reason why this could be of interest would be if I wanted to plot calibration curves for ECMS calibration and gas flux calibrations in one plot. So I might be the only person who's interested in this. Added this as a new issue #123

File: ec_ms_quantification/EC-MS_ixdat_tutorial.ipynb
Comment (AnnaWiniwarter) [reply ec_ms_quantification/EC-MS_ixdat_tutorial.ipynb:349]: I did that, but afterwards realised it was probably not necessary. it shouyld have worked just as well with the original `meas` as imported from zilien. the only reason to use the one with biologic EC data would be because the data is less noisy, but that shouldnt make much of a difference here. Anyway, it's done.

File: ec_ms_quantification/EC-MS_ixdat_tutorial.ipynb
Comment (AnnaWiniwarter) [reply ec_ms_quantification/EC-MS_ixdat_tutorial.ipynb:505]: agreed. a tutorial using siq will follow. 
and see issue#122 for a discussion on the future of `MSInlet`

File: ec_ms_quantification/EC-MS_ixdat_tutorial.ipynb
Comment (AnnaWiniwarter) [reply ec_ms_quantification/EC-MS_ixdat_tutorial.ipynb]: I was using an outdated version of Jupyter (and matplotlib) when I first made the tutorial and I'm slowly realising that a number of the limitations I'm suggesting workarounds for are due to that..

File: ec_ms_quantification/EC-MS_ixdat_tutorial.ipynb
Comment (AnnaWiniwarter) [reply ec_ms_quantification/EC-MS_ixdat_tutorial.ipynb:944]: definitely! will do so for the next version of the notebook.

**Comment by AnnaWiniwarter** [comment]
Made corrections and responded to comments. 
I generally think it's a good idea to split the tutorial in three (or more?) parts, however, as discussed elsewhere, I propose following steps on how to continue with this tutorial:
1)Merge as soon as the small issues are fixed to get a functioning version online ASAP.
2)Separate the tutorial into several smaller ones on "importing", "calibration" and "plotting" as suggested, with two different tutorials for calibration - one with ixdat and one with siq
3)Once the new way of doing native ixdat calibrations is merged into main, update the calibration part in the tutorial.

File: ec_ms_quantification/EC-MS_ixdat_tutorial.ipynb
Comment (ScottSoren) [reply ec_ms_quantification/EC-MS_ixdat_tutorial.ipynb:149]: Agreed on both points. But I meant that the user might be interested in file size reduction with time_step for their own data. In the text block above, you motivate exporting with the ability to make an email-able file.

File: ec_ms_quantification/EC-MS_ixdat_tutorial.ipynb
Comment (ScottSoren) [reply ec_ms_quantification/EC-MS_ixdat_tutorial.ipynb:109]: Nice :)

**Review by ScottSoren — APPROVED** [review]
Looks good! :) 
[This comment](https://github.com/ixdat/tutorials/pull/5/files#r1345681534) is the only one where I had a response to your response.
I'll merge after the meeting and try and get it looking good in the documentation site.

File: ec_ms_quantification/EC-MS_ixdat_tutorial.ipynb
Comment (AnnaWiniwarter) [reply ec_ms_quantification/EC-MS_ixdat_tutorial.ipynb:149]: ahh, yes, that makes sense. should be included in a separate tutorial on importing data! :)

**Pull request merged**
